lib/src/_shared/DateTextField.d.ts

```diff
@@ -13,6 +13,7 @@ export interface DateTextFieldProps extends Omit<TextFieldProps, 'onChange' | 'v
     maxDate?: DateType;
     maxDateMessage?: React.ReactNode;
     mask?: any;
+    onBlur?: () => void;
```

Comment:
actually the typing is (e: SyntheticEvent) => void. 
But maybe there already typing for onBlur in TextField props. Can you test (there is specific typescript usage tests in the project)?